operator/v1/types_machineconfiguration.go

```diff
@@ -56,3 +64,94 @@ type MachineConfigurationList struct {
 	// Items contains the items
 	Items []MachineConfiguration `json:"items"`
 }
+
+type ManagedBootImages struct {
+	// machineManagers is an array of machineManager objects.
+	// The MCO will watch for changes to this list and enroll/un-enroll machine management resources. An entry in this list
+	// consists of the resource type, the API group that the resource belongs to and a selection filter on the resources.
+	// All fields are required.
+	//
+	// Warning: Only one entry per machine management resource type is allowed. If multiple entries target the same type of machine management
+	// resource, all entries associated with that resource type will be ignored. For example, if one entry describes that MAPI machinesets are
+	// disabled for boot image updates and a later entry describes that MAPI machinesets are enabled for boot image updates, both entries
+	// will be ignored.
+	//
+	// +kubebuilder:validation:Required
+	MachineManagers []MachineManager `json:"machineManagers"`
```

Comment:
Should there be a unique `resource/apigroup` pair in this list?

Do you need more than one 
```
- resource: machinesets
  apiGroup: machine.openshift.io
  ...
```
Should the partial matchexpressions not cater for this?

Reply:
I had the same concern when I was working this out, I wrote up some ideas about other ways[here](https://github.com/openshift/api/pull/1672#discussion_r1457860390)